NPE when unitTest variant is disabled for a project

Android Gradle plugin 4.1 and up has the option of completely removing the unitTest variant for an Android project, as exemplified [here](https://github.com/android/gradle-recipes/blob/d63881259df5ed147aecd2a83cad718999001a01/BuildSrc/testVariantFilteringOnBuildType/buildSrc/src/main/kotlin/CustomPlugin.kt#L34).

If you use it, the Pitest plugin will fail to apply, because `TestedVariant.getUnitTestVariant()` returns `null` (in accordance with its method contract):

```
Caused by: java.lang.NullPointerException: Cannot get property 'javaCompileProvider' on null object	
at pl.droidsonroids.gradle.pitest.PitestPlugin.getJavaCompileTask(PitestPlugin.groovy:81)	
at pl.droidsonroids.gradle.pitest.PitestPlugin$getJavaCompileTask.callStatic(Unknown Source)	
at pl.droidsonroids.gradle.pitest.PitestPlugin$_configureTaskDefault_closure7.doCall(PitestPlugin.groovy:201)	
at pl.droidsonroids.gradle.pitest.PitestPlugin.configureTaskDefault(PitestPlugin.groovy:172)	
at org.gradle.internal.metaobject.BeanDynamicObject$MetaClassAdapter.invokeMethod(BeanDynamicObject.java:4
```

This is because [PitestPlugin](https://github.com/koral--/gradle-pitest-plugin/blob/95e0d062d1a954ee571afdf1c22035551683c0b3/src/main/groovy/pl/droidsonroids/gradle/pitest/PitestPlugin.groovy#L201) will unconditionally try to use the `unitTest` variant.

--- a/src/main/groovy/pl/droidsonroids/gradle/pitest/PitestPlugin.groovy
+++ b/src/main/groovy/pl/droidsonroids/gradle/pitest/PitestPlugin.groovy
@@ -198,8 +198,10 @@ class PitestPlugin implements Plugin<Project> {
             from(project.files("${project.buildDir}/intermediates/java_res/${variant.dirName}UnitTest/out"))
             from(project.files("${project.buildDir}/intermediates/unitTestConfig/test/${variant.dirName}"))
             if (variant instanceof TestedVariant) {
-                from(getJavaCompileTask(variant.unitTestVariant).classpath)
-                from(project.files(getJavaCompileTask(variant.unitTestVariant).destinationDir))
+                variant.unitTestVariant?.with { unitTestVariant ->
+                    from(getJavaCompileTask(unitTestVariant).classpath)
+                    from(project.files(getJavaCompileTask(unitTestVariant).destinationDir))
+                }
             }
             from(getJavaCompileTask(variant).classpath)
             from(project.files(getJavaCompileTask(variant).destinationDir))

